scalapb-runtime-grpc/src/main/scala/com/trueaccord/scalapb/grpc/ProtoUtils.scala

```diff
@@ -0,0 +1,25 @@
+package com.trueaccord.scalapb.grpc
+
+import com.google.protobuf.InvalidProtocolBufferException
+import com.trueaccord.scalapb.GeneratedMessage
+import io.grpc.Metadata
+
+object ProtoUtils {
+  def metadataMarshaller[T <: GeneratedMessage](instance: T): Metadata.BinaryMarshaller[T] =
```

Comment:
You can solve this with typeclasses: `implicit companion: GeneratedMessageCompanion[T]`

I don't think it's a good idea to pass instance around. 

Even better, ScalaPbMetadataMarshaller should be an abstract class (because traits can't have parameters yet).